core/vm/environment.go

```diff
@@ -35,6 +37,7 @@ type (
 
 // Context provides the EVM with auxilary information. Once provided it shouldn't be modified.
 type Context struct {
+	context.Context
```

Comment:
You can always make the EVM struct itself smaller by moving more fields into Context and by making the Context field a pointer. That would resolve potential problems with allocation because Context can be reused.